integrations/hardhat/index.js

```diff
@@ -5,7 +5,11 @@ extendEnvironment((hre) => {
   const { chainId, url: rpcUrl } = hre.network.config;
   if (chainId) {
     if (!sapphire.NETWORKS[chainId]) return;
-  } else if (/sapphire/i.test(rpcUrl)) {
+  } else {
+    if (!/sapphire/i.test(rpcUrl)) {
+      return;
+    }
```

Comment:
```suggestion
    if (!/sapphire/i.test(rpcUrl)) return;
```

Reply:
Thanks for finding and fixing this. I'll publish a new package version once this is merged in.

Reply:
Perfect, thank you for taking time to review this.